stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveData.kt

```diff
@@ -74,10 +76,22 @@ internal class MessageListItemLiveData(
         typingLd?.let { typing -> configTypingChange(typing, currentUser) }
     }
 
+    /**
+     * Emits a value from this [MediatorLiveData] class when either
+     * the user gets updated or the deleted message visibility gets
+     * changed.
+     */
     private fun configMessagesChange(messages: LiveData<List<Message>>, getCurrentUser: LiveData<User?>) {
-        val messagesChange = getCurrentUser.changeOnUserLoaded(messages) { changedMessages, currentUser ->
-            messagesChanged(changedMessages, currentUser!!.id)
-        }
+        val messagesChange = getCurrentUser
+            .combineWith(deletedMessageVisibility) { user, _ ->
```

Comment:
This part will re-emit the user when visibility changes unnecessarily but that should not present much overhead since these event should rarely.

It could be re-written in such a way that we use `.combineWith` with `messages` but that may slightly reduce readability.